### PR TITLE
Retryable messages `fuel-vm` part

### PR DESCRIPTION
**Pull request by xgreenx**
`fuel-vm` part of the https://github.com/FuelLabs/fuel-core/issues/946

- Removed the `message_id` field from the message.
- Removed `Output::Message`.
- Re-worked messages to be spendable as a common coin and retryable.
- Added rules regarding `sum_inputs` and `sum_data_messages`. Updated the logic of the VM to only use `SumDataMessages` for `RuntimeBalances`.

### CODE REVIEW

File: fuel-vm/src/interpreter/blockchain.rs [hunk 1]
```diff
@@ -667,6 +666,11 @@ pub(crate) fn timestamp(
     inc_pc(pc)
 }
 
+// TODO: Register b is the address of the begin of the data and register c
```
Comment (xgreenx) [inline fuel-vm/src/interpreter/blockchain.rs:669]: It is a bug and we need to fix it in the follow-up PR

File: fuel-tx/src/transaction/types/input.rs [hunk 2]
```diff
@@ -63,14 +63,16 @@ impl AsField<Vec<u8>> for Vec<u8> {
...
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum Input {
     CoinSigned(CoinSigned),
     CoinPredicate(CoinPredicate),
     Contract(Contract),
-    MessageSigned(MessageSigned),
-    MessagePredicate(MessagePredicate),
+    DepositCoinSigned(DepositCoinSigned),
+    DepositCoinPredicate(DepositCoinPredicate),
+    MetadataSigned(MetadataSigned),
```
Comment (Voxelot) [inline fuel-tx/src/transaction/types/input.rs:74]: Why are we changing the name from message to metadata? We don't have anything in the specs referring to metadata inputs so this is confusing.

Comment (Voxelot) [reply fuel-tx/src/transaction/types/input.rs]: changed to MessageCoin_ and MessageData_

File: fuel-vm/src/interpreter/diff/tests.rs [hunk 3]
```diff
@@ -204,11 +202,12 @@ fn reset_vm_memory() {
 
 #[test]
 fn reset_vm_txns() {
```
Comment (Voxelot) [inline fuel-vm/src/interpreter/diff/tests.rs:204]: this test should clarify what is being tested about reset

**Review by Voxelot — APPROVED** [review]

**Pull request merged**